File: app/src/main/java/com/coinninja/coinkeeper/util/currency/Currency.java
package com.coinninja.coinkeeper.util.currency;

import android.os.Parcelable;

import java.math.BigDecimal;

public interface Currency extends Parcelable {

    String getSymbol();

    String getFormat();

    String getCurrencyFormat();

    void setCurrencyFormat(String format);

    int getMaxNumSubValues();

    int getMaxNumWholeValues();

    String getIncrementalFormat();

    long toLong();

    String toFormattedString();

    String toFormattedCurrency();

    BigDecimal toBigDecimal();

    BTCCurrency toBTC(Currency conversionValue);

    USDCurrency toUSD(Currency conversionValue);

    boolean update(String formattedValue);

    boolean isValid();

    boolean validate(String candidate);

    boolean isZero();

    boolean isCrypto();

    boolean isFiat();
}

File: common/src/main/java/app/dropbit/commons/currency/Currency.kt
package app.dropbit.commons.currency

import android.os.Parcelable
import java.math.BigDecimal

interface Currency : Parcelable {
    val isZero: Boolean
    val isCrypto: Boolean
    val isFiat: Boolean
    val symbol: String
    val format: String
    var currencyFormat: String
    val maxNumSubValues: Int
    val maxNumWholeValues: Int
    val incrementalFormat: String

    fun isValid(): Boolean
    fun toLong(): Long
    fun zero()
    fun toFormattedString(): String
    fun toFormattedCurrency(): String
    fun toBigDecimal(): BigDecimal
    fun toBTC(conversionValue: Currency): BTCCurrency
    fun toUSD(conversionValue: Currency): USDCurrency
    fun toSats(conversionValue: Currency): SatoshiCurrency
    fun update(formattedValue: String): Boolean
    fun validate(candidate: String): Boolean
}